### ISSUE TITLE
RpcTestWithFork.test_nested_rref is flaky in CI

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#29382]
CI build: pytorch_linux_xenial_cuda9_cudnn7_py3_NO_AVX2_test

```
Nov 07 15:14:02 ======================================================================
Nov 07 15:14:02 ERROR: test_nested_rref (__main__.RpcTestWithFork)
Nov 07 15:14:02 ----------------------------------------------------------------------
Nov 07 15:14:02 Traceback (most recent call last):
Nov 07 15:14:02   File "/var/lib/jenkins/workspace/test/common_distributed.py", line 130, in wrapper
Nov 07 15:14:02     self._join_processes(fn)
Nov 07 15:14:02   File "/var/lib/jenkins/workspace/test/common_distributed.py", line 211, in _join_processes
Nov 07 15:14:02     self._check_return_codes(elapsed_time)
Nov 07 15:14:02   File "/var/lib/jenkins/workspace/test/common_distributed.py", line 230, in _check_return_codes
Nov 07 15:14:02     raise RuntimeError('Process {} terminated or timed out after {} seconds'.format(i, elapsed_time))
Nov 07 15:14:02 RuntimeError: Process 0 terminated or timed out after 300.2084217071533 seconds
Nov 07 15:14:02 
Nov 07 15:14:02 ----------------------------------------------------------------------
Nov 07 15:14:02 Ran 53 tests in 305.101s
Nov 07 15:14:02 
Nov 07 15:14:02 FAILED (errors=1, skipped=5)

```

https://app.circleci.com/jobs/github/pytorch/pytorch/3510105

@mrshenli @aazzolini

cc @ezyang @gchanan @zou3519 @jerryzh168 @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @rohan-varma @xush6528

**Comment by mrshenli** [comment]
https://app.circleci.com/jobs/github/pytorch/pytorch/3515689

Hit the same error:

```
Nov 07 23:04:05 test_nested_remote (__main__.RpcTestWithFork) ... OMP: Error #13: Assertion failure at kmp_runtime.cpp(1407).
Nov 07 23:04:05 OMP: Hint Please submit a bug report with this message, compile and run commands used, and machine configuration info including native compiler and operating system versions. Faster response will be obtained by including all program sources. For information on submitting this issue, please see http://www.intel.com/software/products/support/.
Nov 07 23:09:05 ERROR
```

AFAIK, RRef/RPC does not use OMP, although `torch.add` might, not sure it would do so with a 2X2 tensor. 

@ezyang @VitalyFedyunin Have you seen similar OMP error before? I searched a bit, and discussions on other similar errors seems to suggest setting KMP_INIT_AT_FORK=FALSE. Any comments?

**Comment by ezyang** [comment]
This error looks similar to the error in this PR https://github.com/pytorch/pytorch/pull/29006 cc @peterbell10

**Comment by mrshenli** [comment]
Same test failed on a different error. Let me disable it for now.

```
Nov 08 17:21:19 test_nested_rref (__main__.RpcTestWithFork) ... terminate called after throwing an instance of 'gloo::IoException'
Nov 08 17:21:19   what():  [/var/lib/jenkins/workspace/third_party/gloo/gloo/transport/tcp/pair.cc:563] Read error [172.17.0.2]:43703: Connection reset by peer
Nov 08 17:24:39 ERROR
```

**Comment by mruberry** [comment]
Yet another example: https://app.circleci.com/jobs/github/pytorch/pytorch/3529088

**Comment by mrshenli** [comment]
https://app.circleci.com/jobs/github/pytorch/pytorch/3570825

```
Nov 12 17:17:21 test_rpc_return_rref (__main__.RpcTestWithFork) ... OMP: Error #13: Assertion failure at kmp_csupport.cpp(590).
Nov 12 17:17:21 OMP: Hint Please submit a bug report with this message, compile and run commands used, and machine configuration info including native compiler and operating system versions. Faster response will be obtained by including all program sources. For information on submitting this issue, please see http://www.intel.com/software/products/support/.
Nov 12 17:22:21 ERROR
```

**Comment by ezyang** [comment]
This is still affecting me https://app.circleci.com/jobs/github/pytorch/pytorch/3615514

**Comment by mrshenli** [comment]
@ezyang The error above does not seem to relate to RPC?

```
Nov 15 17:53:37 Running test_cuda ... [2019-11-15 17:53:37.137265]
Nov 15 17:53:40 Traceback (most recent call last):
Nov 15 17:53:40   File "test_cuda.py", line 92, in <module>
Nov 15 17:53:40     class TestCuda(TestCase):
Nov 15 17:53:40   File "test_cuda.py", line 1815, in TestCuda
Nov 15 17:53:40     @unittest.expectedFailure()
Nov 15 17:53:40 TypeError: expectedFailure() missing 1 required positional argument: 'test_item'
Nov 15 17:53:40 Traceback (most recent call last):
Nov 15 17:53:40   File "test/run_test.py", line 454, in <module>
Nov 15 17:53:40     main()
Nov 15 17:53:40   File "test/run_test.py", line 447, in main
Nov 15 17:53:40     raise RuntimeError(message)
Nov 15 17:53:40 RuntimeError: test_cuda failed!
```

**Comment by mrshenli** [comment]
closed by #29827
